FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt

```diff
@@ -149,30 +149,32 @@ class Flagsmith constructor(
             result(res.map { it.traits })
         }.also { lastUsedIdentity = identity }
 
-    fun setTrait(trait: Trait, identity: String, result: (Result<TraitWithIdentity>) -> Unit) =
-        retrofit.postTraits(IdentityAndTraits(identity, listOf(trait)))
+    fun setTrait(trait: Trait, identity: String, transient: Boolean = false, result: (Result<TraitWithIdentity>) -> Unit) =
+        retrofit.postTraits(IdentityAndTraits(identity, listOf(trait), transient))
             .enqueueWithResult(result = {
                 result(it.map { response -> TraitWithIdentity(
                     key = response.traits.first().key,
                     traitValue = response.traits.first().traitValue,
-                    identity = Identity(identity)
+                    identity = Identity(identity),
+                    transient = response.traits.first().transient
                 )})
             })
 
-    fun setTraits(traits: List<Trait>, identity: String, result: (Result<List<TraitWithIdentity>>) -> Unit) {
-        retrofit.postTraits(IdentityAndTraits(identity, traits)).enqueueWithResult(result = {
+    fun setTraits(traits: List<Trait>, identity: String, transient: Boolean, result: (Result<List<TraitWithIdentity>>) -> Unit) {
```

Comment:
Hmm... this only allows you to set all traits in this call as transient, right? The main use case here is to only set certain traits as transient. Is there a way that we can make that work without breaking the interface?

Reply:
Corrected